### PR TITLE
Feat: Make track list static

### PR DESCRIPTION
**Pull request by greenbech**
The track list is now placed left to the piano roll by a row flex box. This is better since it enables quick navigation between the track and makes it easier to mute/solo a given track.

The ghost tracks are now global instead of being customizeable on each track. The former method was more flexible, but quite cumbersome to use. All ghost tracks can be toogled with `alt` + `leftMouseButton`.

The track list drawer is now always opened when a new file is opened.

I'd like a thorough review since I'm still not very used to React and frond end work

### CODE REVIEW

**Comment by ryohey** [comment]
It's so intuitive! I will review it later, so please wait a while. Thanks.

File: src/main/components/PianoRollToolbar/VolumeSlider.tsx [hunk 1]
```diff
@@ -33,7 +33,9 @@ export interface VolumeSliderProps {
 
 const _VolumeSlider: FC<VolumeSliderProps> = observer(({ trackId }) => {
   const rootStore = useStores()
-  const volume = rootStore.pianoRollStore.currentVolume
+  const volume = rootStore.song.tracks[trackId].getVolume(
+    rootStore.services.player.position
+  )
```
Comment (ryohey) [inline src/main/components/PianoRollToolbar/VolumeSlider.tsx:38]: Is this necessary change?

File: src/main/components/PianoRoll/PianoRollEditor.tsx [hunk 2]
```diff
@@ -1,21 +1,36 @@
...
   display: flex;
   flex-direction: column;
   flex-grow: 1;
 `
 
-export const PianoRollEditor: FC = () => {
+const RowContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-grow: 1;
```
Comment (ryohey) [inline src/main/components/PianoRoll/PianoRollEditor.tsx:19]: When there are many tracks in the list, view size will get bigger than window.
We should make TrackList scrollable.

image

Add styles to restrict RowContainer`s height.

```suggestion
  flex-grow: 1;
  overflow: hidden;
  flex-basis: 0;
```

Then add overflow style to TrackList.
https://github.com/ryohey/signal/blob/c5f56726066334cdba76d7f90f7decdde7c6cdb2/src/main/components/TrackList/TrackList.tsx#L86

```diff
-<List>
+<List style={{ overflowY: "auto" }}>
```

image

**Review by ryohey — CHANGES_REQUESTED** [review]
Thank you for your contribution!
I noticed some small issues. I'm happy if you could fix that.

File: src/main/components/PianoRollToolbar/VolumeSlider.tsx (hunk 1 above)
Comment (greenbech) [reply src/main/components/PianoRollToolbar/VolumeSlider.tsx:38]: I originally tried to add the volume slider to the track list, but my first attempt was quite buggy so I decided to drop it. I also wanted it to be global for the entire track—a percent between 0-100% that would be multiplyed with the tick spesific volume—but wasn't able to implement it easily. Can open an issue for this and drop the change :^)

File: src/main/components/PianoRoll/PianoRollEditor.tsx (hunk 2 above)
Comment (greenbech) [reply src/main/components/PianoRoll/PianoRollEditor.tsx:19]: Great catch! Didn't test it with any songs that had that many tracks before you mentioned it. Updated based on your comments now. I must say that I don't think that scroll bar looks very pretty (I would be nicer if it were more themed like the other scroll bars), but I guess the style can be updated in a new PR

**Review by ryohey — APPROVED** [review]
LGTM! Thanks a lot!

**Comment by Endermaster3690** [comment]
#100 pog

**Pull request merged**